### PR TITLE
[BREAKING] Register extbase type converters as services

### PR DESCRIPTION
**Pull request by linawolf**
I will add an elaborate example in a follow up.

Needs to be backported manually to 11.5

Releases: main

resolves https://github.com/TYPO3-Documentation/Changelog-To-Doc/issues/36

### CODE REVIEW

File: Documentation/ExtensionArchitecture/Extbase/Reference/Controller/TypeConverter.rst [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+======================
+
+..  versionchanged:: 12.0
+    Starting with TYPO3 12.0 a type converter does not have to be registered
+    via the now deprecated method :php:`\TYPO3\CMS\Extbase\Utility\ExtensionUtility::registerTypeConverter()`.
+    Remove calls to this method when dropping TYPO3 v11 support. It will be
+    removed with TYPO3 13. Register a type converter in your extension's
+    :file:`Services.yaml` instead.
+
+A custom type converter must extend the abstract class
```
Comment (brotkrueml) [inline Documentation/ExtensionArchitecture/Extbase/Reference/Controller/TypeConverter.rst:37]: Nope, must implement the TypeConverterInterface or extend the AbstractTypeConverter.

**Review by brotkrueml — CHANGES_REQUESTED** [review]
Please use either "\" prefixing a namespace or not, but not mixing in the same chapter.

**Comment by brotkrueml** [comment]
Another thing: As we use a DateTime converter as example in the introduction, we should use that at the end of the chapter and not the Boolean converter to be stringent.

File: Documentation/ExtensionArchitecture/Extbase/Reference/Controller/TypeConverter.rst [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+===============
+Type converters
+===============
+
+Type converters are commonly used when it is necessary to convert from one type
+into another. They are usually applied in the controller in the method
+:php:`initialize<actionName>Action()` method.
+
+For example a date might be given as string in some language,
+:php:`"Freitag, 7. Oktober 2022"` or as UNIX time stamp: :php:`1665159559`.
```
[the remark this answers is not in the document]
Comment (linawolf) [reply Documentation/ExtensionArchitecture/Extbase/Reference/Controller/TypeConverter.rst:15]: I made this as an example for a string that cannot be resolved automatically and would therefore need custom handling

[the remark this answers is not in the document]
Comment (brotkrueml) [reply Documentation/ExtensionArchitecture/Extbase/Reference/Controller/TypeConverter.rst:15]: Yup, but we should use English - some person may not recognize this example as a date string.

**Review by brotkrueml — APPROVED** [review]

**Pull request merged**